classes/controller/FrontController.php

```diff
@@ -1949,6 +1950,45 @@ protected function makeAddressForm()
         return $form;
     }
 
+    /**
+     * Get customer login form
+     *
+     * @return CustomerLoginForm
+     */
+    public function getLoginForm()
+    {
+        return $this->makeLoginForm()
+            ->fillWith(Tools::getAllValues())
+            ->setAction($this->context->link->getPageLink('authentication'))
+        ;
+    }
+
+    /**
+     * Get customer register form
+     *
+     * @return CustomerForm
+     */
+    public function getRegisterForm()
+    {
+        return $this->makeCustomerForm()
+            ->fillWith(Tools::getAllValues())
+            ->setAction($this->context->link->getPageLink('authentication&create_account=1'))
```

Comment:
```suggestion
            ->setAction($this->context->link->getPageLink('authentication&create_account=1'))
```

I think you should pass it as a parameter, I think this version would be problematic for translated URLs?

Reply:
the `getPageLink` method will take `$idLang` from context if its not given as the third parameter

maybe having an optional $idLang parameter would be enough ? no need to use it here, and if for some reason someone needs to get the link for another language, he can pass an `idLang` as a parameter.

Reply:
second parameter `$ssl` can be set to null, it will be taken from the `Configuration` in the `getBaseLink` method

Reply:
@matthieu-rolland I meant the 4th parameter, `$request`

`getPageLink('authentication', true, null, ['create_account' => 1])`

Reply:
ah I misunderstood your point ! 

you're talking about `&create_account=1` ? yes it seems like this parameter is stripped and unused in the the `getPageLink` method, seems like it should passed in the `$request` parameter ?

Reply:
but from what I see now it should be handled properly:

```
//If $controller contains '&' char, it means that $controller contains request data and must be parsed first
        $p = strpos($controller, '&');
        if ($p !== false) {
            $request = substr($controller, $p + 1);
            $requestUrlEncode = false;
            $controller = substr($controller, 0, $p);
        }
```

so it should be all good

Reply:
yes just saw it too ;)

I missed your point at first, but yes should be all good 👍

Reply:
hummm ...
Looking  https://github.com/PululuK/PrestaShop/blob/add-global-forms-vars/classes/Link.php#L1116
Finaly `getPageLink('authentication', true, null, ['create_account' => 1])` = `getPageLink('authentication&create_account=1')`

Reply:
I don't really like this kind of method where the same thing can be achieved in different ways, it keeps the code from being consistent...  but we have to deal with it unfortunately